File: Data Structures/Linear data structures/stack/c/stackLinkedlist.c
/* C program to Implement a stack using singly linked list. */
#include <stdio.h>
#include <stdlib.h>

struct Node
{
    int data;
    struct Node *link;
};
struct Node *top;

void push(int data)
{
    struct Node *temp;
    temp = malloc(sizeof(struct Node));
    if (!temp)
    {
        printf("stack overflow");
        exit(1);
    }
    temp->data = data;
    temp->link = top;
    top = temp;
}

int isEmpty()
{
    return top == NULL;
}

void pop()
{
    struct Node *temp;
    if (isEmpty())
    {
        printf("stack underflow");
        exit(1);
    }
    else
    {
        temp = top;
        top = top->link;
        free(temp);
    }
}
int peek()
{
    if (!isEmpty())
    {
        return top->data;
    }
    else
        exit(1);
}

void display()
{
    struct Node *temp;
    if (isEmpty())
    {
        printf("stack underflow");
        exit(1);
    }
    else
    {
        temp = top;
        while (temp != NULL)
        {
            printf("%d ", temp->data);
            temp = temp->link;
        }
    }
}

int main()
{
    push(3);
    push(236);
    push(15);

    display();

    pop();
    printf("\n stack after pop ");

    display();
    printf("\ngettop ");
    peek();
}

File: Data Structures/Linear data structures/stack/python/stackLinkedlist.py
# Python program for linked list implementation of stack
from logging import root


class StackNode:
    # Constructor to initialize a node
    def __init__(self, data):
        self.data = data
        self.next = None


class Stack:
    # Constructor to initialize a topNode node
    def __init__(self):
        self.topNode = None

    def isEmpty(self):
        return True if self.topNode is None else False

    def push(self, data):
        newNode = StackNode(data)
        newNode.next = self.topNode
        self.topNode = newNode

    def pop(self):
        if self.isEmpty():
            return float("-inf")
        temp = self.topNode
        self.topNode = self.topNode.next
        poped = temp.data
        return poped

    def peek(self):
        if self.isEmpty():
            return float("-inf")
        return self.topNode.data


stack = Stack()
stack.push(3)
stack.push(23)
stack.push(6)
print(stack)
